evalml/pipelines/components/transformers/scalers/standard_scaler.py

```diff
@@ -82,5 +83,7 @@ def fit_transform(self, X, y=None):
         Returns:
             pd.DataFrame: Transformed data.
         """
-        X = infer_feature_types(X)
+        if not isinstance(X, pd.DataFrame):
+            X = infer_feature_types(X)
+        X = X.select_dtypes(exclude=["datetime"])
```

Comment:
Saves a call to `infer_feature_types` since the call is also made in `fit` and `transform`